src/OpenLoco/Map/TileManager.cpp

```diff
@@ -549,6 +549,64 @@ namespace OpenLoco::Map::TileManager
         _numAnimations = 0;
     }
 
+    // 0x004C5596
+    uint16_t countSurroundingWaterTiles(Map::Pos2 pos)
+    {
+        pos.x = (pos.x & 0xFFE0) - 0xA0;
+        pos.y = (pos.y & 0xFFE0) - 0xA0;
+
+        uint16_t surroundingWaterTiles = 0;
+        for (uint8_t yOffset = 0; yOffset < 11; yOffset++)
+        {
+            for (uint8_t xOffset = 0; xOffset < 11; xOffset++)
+            {
+                auto tile = get(Map::Pos2(pos.x + xOffset, pos.y + yOffset));
+                for (auto& element : tile)
+                {
+                    auto* surface = element.asSurface();
+                    if (surface == nullptr)
+                        continue;
+
+                    if (surface->water() > 0)
+                        surroundingWaterTiles++;
+
+                    break;
+                }
+            }
+        }
+
+        return surroundingWaterTiles;
+    }
+
+    // 0x004BE048
+    uint16_t countSurroundingTrees(Map::Pos2 pos)
+    {
+        pos.x = (pos.x & 0xFFE0) - 0xA0;
+        pos.y = (pos.y & 0xFFE0) - 0xA0;
+
+        uint16_t surroundingTrees = 0;
+        for (uint8_t yOffset = 0; yOffset < 11; yOffset++)
+        {
+            for (uint8_t xOffset = 0; xOffset < 11; xOffset++)
+            {
+                auto tile = get(Map::Pos2(pos.x + xOffset, pos.y + yOffset));
+                for (auto& element : tile)
+                {
+                    auto* tree = element.asTree();
```

Comment:
vanilla was more complex than this making sure that the tree element was above the surface element. Not sure if we should bother recreating that.